File: spring 2/FE Spring 2/library/src/app/model/export.ts
import {Pigsty} from './pigsty';
import {Employee} from './employee';

export interface Export {
  id?: number;
  codeExport?: string;
  company?: string;
  saleDate?: string;
  amount?: number;
  kilogram?: number;
  totalMoney?: number;
  price?: number;
  typePigs?: number;
  isDeleted?: boolean;
  pigstyDto?: Pigsty;
  employDto?: Employee;
  codeEmployee?: string;
  nameEmployee?: string;
}

File: spring 2/BE Spring 2/library/src/main/java/vn/codegym/pig_farm/entity/Export.java
package vn.codegym.pig_farm.entity;


import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.time.LocalDate;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Export {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(columnDefinition = "VARCHAR(50)")
    private String codeExport;
    @Column(columnDefinition = "VARCHAR(50)")
    private String company;

    private Double kilogram;

    private Integer amount;

    private Double price;

    @Column(columnDefinition = "BIT(1) DEFAULT 0")
    private Boolean isDeleted;

    private Integer typePigs;

    @Column(columnDefinition = "DATE")
    private LocalDate saleDate;

    @ManyToOne
    @JoinColumn(name = "pigsty_id", referencedColumnName = "id")
    private Pigsty pigsty;

    @ManyToOne
    @JoinColumn(name = "employee_id", referencedColumnName = "id")
    private Employee employee;
}
